ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/api/WriteServiceWithDeduplicationSupportSpec.scala

```diff
@@ -121,4 +121,58 @@ class WriteServiceWithDeduplicationSupportSpec
       })
   }
 
+  "use actAs and readAs as readers for the completion stream" in {
+    val submitterInfoWithParties = submitterInfo.copy(
+      actAs = List(Ref.Party.assertFromString("party1")),
+      readAs = List(Ref.Party.assertFromString("party2")),
+    )
+    when(
+      mockDeduplicationPeriodSupport.supportedDeduplicationPeriod(
+        any[DeduplicationPeriod],
+        any[Duration],
+        any[LedgerTimeModel],
+        any[ApplicationId],
+        any[Set[Ref.Party]],
+        any[Instant],
+      )(
+        any[Materializer],
+        any[ExecutionContext],
+        any[LoggingContext],
+        any[ContextualizedErrorLogger],
+      )
+    ).thenReturn(Future.successful(submitterInfo.deduplicationPeriod))
+    when(
+      mockWriteService.submitTransaction(
+        any[SubmitterInfo],
+        any[TransactionMeta],
+        any[SubmittedTransaction],
+        any[Long],
+      )(any[LoggingContext], any[TelemetryContext])
+    ).thenReturn(Future.successful[SubmissionResult](Acknowledged).asJava)
+    service
+      .submitTransaction(
+        submitterInfo,
+        transactionMeta,
+        submittedTransaction,
+        0,
+      )
+      .asScala
+      .map(_ => {
+        verify(mockDeduplicationPeriodSupport).supportedDeduplicationPeriod(
+          any[DeduplicationPeriod],
+          any[Duration],
+          any[LedgerTimeModel],
+          any[ApplicationId],
+          eqTo((submitterInfoWithParties.readAs ++ submitterInfoWithParties.actAs).toSet),
```

Comment:
Can you pull this `Set[Party]` out as a variable near the top so it's easier to understand?